src/bootstrap/src/core/builder/tests.rs

```diff
@@ -261,8 +261,14 @@ fn ci_rustc_if_unchanged_logic() {
     // Make sure "if-unchanged" logic doesn't try to use CI rustc while there are changes
     // in compiler and/or library.
     if config.download_rustc_commit.is_some() {
-        let has_changes =
-            config.last_modified_commit(&["compiler", "library"], "download-rustc", true).is_none();
+        let mut paths = vec!["compiler"];
+
+        // Handle library tree the same way as in `Config::download_ci_rustc_commit`.
```

Comment:
Suggestion: I think we should either:

- Have a single source of truth for the paths calculation fed to `last_modified_commit` for this purpose,
- Or also add a comment at the source site where we also do this paths adjustment.

wdyt?

Reply:
> Have a single source of truth for the paths calculation fed to last_modified_commit for this purpose,

We can't do that as one side excludes those paths and the other includes them. Also, using that approach would cause this coverage to miss the expected behaviour (imagine that function somehow have bugs in its logic).

> Or at least add a comment at the source site where we also do this paths adjustment to remind also updating this test.

Sure, let me know exactly where and what you want me to write.

Reply:
Maybe just cc this test next to https://github.com/rust-lang/rust/blob/master/src%2Fbootstrap%2Fsrc%2Fcore%2Fconfig%2Fconfig.rs#L2984-L2990?